e2e/test/iothub/twin/TwinE2ETests.cs

```diff
@@ -654,6 +656,58 @@ await deviceClient
             Twin serviceTwin = await serviceClient.Twins.GetAsync(testDevice.Id).ConfigureAwait(false);
             Assert.IsFalse(serviceTwin.Properties.Reported.Contains(propName1));
         }
+
+        [DataTestMethod, Timeout(LongRunningTestTimeoutMilliseconds)]
+        [DataRow(IotHubClientTransportProtocol.Tcp)]
+        [DataRow(IotHubClientTransportProtocol.WebSocket)]
+        [TestCategory("LongRunning")]
+        public async Task Twin_Client_SetETag_Works(IotHubClientTransportProtocol transportProtocol)
```

Comment:
This test is failing with the error:
Newtonsoft.Json.JsonWriterException: Unsupported type: Azure.ETag. Use the JsonSerializer class to get the object's JSON representation. Path ''.

Any suggestions for it?

Reply:
Did you add [this tag](https://github.com/Azure/azure-iot-sdk-csharp/blob/7a9ccb4d95c5ff65d811563a8696f1b82f70ae9d/iothub/service/src/Registry/Models/Device.cs#L55) to your ETag field that fails to serialize?